### PR TITLE
git-subrepo: init at 0.4.0

### PR DESCRIPTION
**Pull request by ryantrinkle**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] N/A (nothing depends on this) Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchFromGitHub, git, which }:
+
+stdenv.mkDerivation rec {
+  name = "git-subrepo-${version}";
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]:
```suggestion
  pname = "git-subrepo";
```

File: pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "ingydotnet";
+    repo = "git-subrepo";
+    rev = "${version}";
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]:
```suggestion
    rev = version;
```

File: pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    sha256 = "05m2dm9gq2nggwnxxdyq2kjj584sn2lxk66pr1qhjxnk81awj9l7";
+  };
+
+  buildInputs = [
+    git which
+  ];
+
+  makeFlags = [
+    "PREFIX=$(out)"
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]:
```suggestion
    "PREFIX=${placeholder ''out''}"
```

File: pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "INSTALL_LIB=$(out)/bin"
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]:
```suggestion
    "INSTALL_LIB=${placeholder ''out''}/bin"
```

File: pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "INSTALL_MAN=$(out)/share/man/man1"
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]:
```suggestion
    "INSTALL_MAN=${placeholder ''out''}/share/man/man1"
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ingydotnet";
+    repo = "git-subrepo";
+    rev = "${version}";
+    sha256 = "05m2dm9gq2nggwnxxdyq2kjj584sn2lxk66pr1qhjxnk81awj9l7";
+  };
+
+  buildInputs = [
+    git which
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]: Is `which` actually used in the program or for during the build?

Comment (ryantrinkle) [reply pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]: Yep, surprisingly! https://github.com/ingydotnet/git-subrepo/blob/5d6aba91dbff3157e498b0a795e99e2fcb7d9ec4/Makefile#L2

Comment (worldofpeace) [reply pkgs/applications/version-management/git-and-tools/git-subrepo/default.nix]: Ah I see. Can you move `which` to `nativeBuildInputs` then?

Also, I guessing this hard depends on the git executables, can you wrap the program like?
```
nativeBuildInputs = [
  makeWrapper
];

postFixup = ''
  wrapProgram $out/bin/$program_name \
    --prefix PATH : "${git}/bin"
'';

**Comment by ryantrinkle** [comment]
@worldofpeace Thanks for the review!  I think everything you mentioned is now covered.

**Review by Ma27 — APPROVED** [review]
Built and tested locally, change seems fine now :+1:

**Pull request merged**